src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,260 @@
+
+        GenotypesContext newGenotypes = subsetAlleles(originalVc.getGenotypes(), originalVc.getAlleles(), allelesToKeep);
+        vcBuilder.genotypes(newGenotypes);
+        return vcBuilder.make();
+    }
+
+    /**
+     * swaps one of the alleles in a VC (and its genotypes) with another.
+     */
+    public static VariantContext swapAlleles(final VariantContext originalVc, final Allele oldAllele, final Allele newAllele) {
+        if (!originalVc.getAlleles().contains(oldAllele)) return originalVc;
+
+        final LinkedList<Allele> alleles = new LinkedList<>(originalVc.getAlleles());
+        alleles.set(alleles.indexOf(oldAllele), newAllele);
+
+        VariantContextBuilder vcBuilder = new VariantContextBuilder(originalVc).alleles(alleles);
+        GenotypesContext newGTs = GenotypesContext.create(originalVc.getGenotypes().size());
+
+        for (final Genotype g : originalVc.getGenotypes()) {
+            if (!g.getAlleles().contains(oldAllele)) {
+                newGTs.add(g);
+            } else {
+                final GenotypeBuilder gb = new GenotypeBuilder(g);
+                final LinkedList<Allele> genotypeAlleles = new LinkedList<>(g.getAlleles());
+                genotypeAlleles.set(genotypeAlleles.indexOf(oldAllele), newAllele);
+                gb.alleles(genotypeAlleles);
+                newGTs.add(gb.make());
+            }
+        }
+        vcBuilder.genotypes(newGTs);
+        return vcBuilder.make();
+    }
+
+    /**
+     * Create the new GenotypesContext with the subsetted PLs and ADs
+     * <p>
+     * Will reorder subsetted alleles according to the ordering provided by the list allelesToKeep
+     *
+     * @param originalGs      the original GenotypesContext
+     * @param originalAlleles the original alleles
+     * @param allelesToKeep   the subset of alleles to use with the new Genotypes
+     * @return a new non-null GenotypesContext
+     */
+    public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
+                                                 final List<Allele> originalAlleles,
+                                                 final List<Allele> allelesToKeep) {
+        nonNull(originalGs, "original GenotypesContext must not be null");
+        nonNull(allelesToKeep, "allelesToKeep is null");
+        nonEmpty(allelesToKeep, "must keep at least one allele");
+        validateTrue(allelesToKeep.get(0).isReference(), "First allele must be the reference allele");
+        validateTrue(allelesToKeep.stream().allMatch(originalAlleles::contains), "OriginalAlleles must contain allelesToKeep");
+
+        int indexOfLast = -1;
+        for (Allele a : allelesToKeep) {
+            validateTrue(indexOfLast < originalAlleles.indexOf(a), "alleles to keep must maintain the order of the original alleles");
+            indexOfLast = originalAlleles.indexOf(a);
+        }
+
+        final int allelesIndex[] = allelesToKeep.stream().mapToInt(originalAlleles::indexOf).toArray();
```

Comment:
it's much clearer to write `int[] allelesIndex` instead of `int allelesIndex[]`

Reply:
done.